src/main/webapp/app/tasks/task-list/task-list.component.ts

```diff
@@ -25,4 +27,22 @@ export class TaskListComponent {
       this.deleted.emit(task);
     });
   }
+  doneTask(task: Task): void {
+    this.taskService.done(task.id).subscribe(() => {
+      this.done.emit(task);
+    });
+  }
+  clearDoneTasks(): void {
+    this.taskService.clear().subscribe(() => {
+      this.cleared.emit();
+    });
+  }
+  showClearButton(): boolean {
+    const inCompleteTasks = this.tasks.filter(task => task.isComplete === true);
```

Comment:
shorter is better:

    const inCompleteTasks = this.tasks.filter(task => task.isComplete);
    return !!inCompleteTasks.length;